runtime/lib/mirrors_impl.dart

```diff
@@ -1676,4 +1681,20 @@ class _Mirrors {
     }
     return typeMirror;
   }
+
+  static Type _resolveType(Type key, List<Type> typeArguments) {
+    if (typeArguments.isEmpty) {
+      throw new ArgumentError.value(
+        typeArguments, 'typeArguments', 'Type arguments list can not be empty.'
+      );
+    }
+    var resolvedTypeMirror = makeLocalTypeMirrorWithTypeArguments(
+      key, typeArguments.toList(growable: false));
+    var actualType = resolvedTypeMirror.reflectedType;
+    // We check if the actual type has been cached already and return cached copy.
+    if (_instantiationCache[actualType] == null) {
```

Comment:
This caching will happen anyway in reflectType, so why just return the type directly from the native?